sources/rest_api/config_setup.py

```diff
@@ -81,11 +83,36 @@ class IncrementalParam(NamedTuple):
     end: Optional[str]
 
 
+@dataclass
+class PaginatorMap:
+    _map: Dict[str, Type[BasePaginator]] = field(
+        default_factory=lambda: copy(PAGINATOR_MAP)
```

Comment:
I'd make the PAGINATOR_MAP private so it can be used only via the PaginatorMap class.